crates/requirements-txt/src/lib.rs

```diff
@@ -294,30 +340,79 @@ pub struct RequirementsTxt {
 }
 
 impl RequirementsTxt {
-    /// See module level documentation
-    #[instrument(skip_all, fields(requirements_txt = requirements_txt.as_ref().as_os_str().to_str()))]
+    #[instrument(skip_all, fields(requirements_txt = %requirements_txt))]
     pub fn parse(
-        requirements_txt: impl AsRef<Path>,
+        requirements_txt: &str,
         working_dir: impl AsRef<Path>,
     ) -> Result<Self, RequirementsTxtFileError> {
-        let content =
-            uv_fs::read_to_string(&requirements_txt).map_err(|err| RequirementsTxtFileError {
-                file: requirements_txt.as_ref().to_path_buf(),
-                error: RequirementsTxtParserError::IO(err),
+        let requirements_txt_source = RequirementsTxtSource::parse(requirements_txt, &working_dir)
+            .map_err(|err| RequirementsTxtFileError {
+                file: PathBuf::from(requirements_txt),
+                error: RequirementsTxtParserError::Url {
+                    source: err,
+                    url: requirements_txt.to_string(),
+                    start: 0,
+                    end: 0,
+                },
             })?;
+        // Get the content of the file or URL
+        let content = match &requirements_txt_source {
+            RequirementsTxtSource::Path(path) => {
+                uv_fs::read_to_string(path).map_err(|err| RequirementsTxtFileError {
+                    file: path.clone(),
+                    error: RequirementsTxtParserError::IO(err),
+                })?
+            }
+            RequirementsTxtSource::Url(url) => {
+                // Use the `reqwest` crate to download the file.
+                reqwest::blocking::get(url.as_str())
+                    .map_err(|err| RequirementsTxtFileError {
+                        file: PathBuf::from(url.as_str()),
+                        error: RequirementsTxtParserError::IO(io::Error::new(
+                            io::ErrorKind::Other,
+                            err.to_string(),
+                        )),
+                    })?
+                    .error_for_status()
+                    .map_err(|err| RequirementsTxtFileError {
+                        file: PathBuf::from(url.as_str()),
+                        error: RequirementsTxtParserError::IO(io::Error::new(
+                            io::ErrorKind::Other,
+                            err.to_string(),
+                        )),
+                    })?
+                    .text()
+                    .map_err(|err| RequirementsTxtFileError {
+                        file: PathBuf::from(url.as_str()),
+                        error: RequirementsTxtParserError::IO(io::Error::new(
+                            io::ErrorKind::Other,
+                            err.to_string(),
+                        )),
+                    })?
```

Comment:
This is the section of code that isn't working - calling`::blocking::get` while a tokio runtime is active